### ISSUE TITLE
[Glue] Integrate Update User Info API with Frontend

### ISSUE THREAD

**Issue by jcsison** [jcsison/dealicious#32]
## Description
Depends on #31. Integrate update user info API with the frontend. The user should be able to change their own name and email in the user settings page. The ability to change the user's profile picture will be deferred to a later issue.

## Acceptance Criteria
- The user is able to change their own name and email in the user settings page
- A loading indicator appears on the "Update Profile" button when the user confirms their changes
- No loading indicator appears and no API call is made if the user makes no changes to their name and email
